### ISSUE TITLE
some suggestion for omnihuman reproduction

### ISSUE THREAD

**Issue by taichuai** [johndpope/OmniHuman-1-hack#6]
I suggest skipping the use of SEAWEED first, taking wan2.1 as the base model, and implementing the audio driver solution first, so that the effect can be easily verified.

**Comment by johndpope** [comment]
im curious if it would work - seaweed claims high speed diffusion with a lot of gpus -
 
<img width="1092" alt="Image" src="https://github.com/user-attachments/assets/04a55633-354a-496a-a410-8f19f5940527" /> 
<img width="1402" alt="Image" src="https://github.com/user-attachments/assets/6c33759f-cca2-4344-b3ab-e88fbc46ce4c" />

i ordered a 5090 - waiting for delivery....hopefully by end of this month or next.....
that will give me a tiny bit more head room with 32gb. 
this t5 text encoder is 11gb - it's almost like i need to throw in another gpu just as dedicated  gpu. k80 goes for like $50 / on ebay. i need an a6000 48gbs - maybe i rent some vertex compute.

**Comment by johndpope** [comment]
some progress on seaweed. just need to throw > 100+ gpus at it.

**Comment by johndpope** [comment]
i push some code for omnihuman without seaweed a few days back. this needs redoing....


     def training_step(
        self,
        frames: torch.Tensor,
        conditions: Dict[str, torch.Tensor],
        t: torch.Tensor
    ) -> torch.Tensor:
        """Compute flow-matching loss for training.
        
        Args:
            frames: Ground truth frames [B, C, T, H, W]
            conditions: Dictionary of condition tokens
            t: Timestep values [B]
            
        Returns:
            Flow matching loss
        """
        logger.debug(f"Training step with frames shape: {frames.shape}")
        
        # Add noise using flow-matching
        noise = torch.randn_like(frames)
        noisy_frames = (1 - t.view(-1, 1, 1, 1, 1)) * frames + t.view(-1, 1, 1, 1, 1) * noise
        
        # Process conditions
        text_context = conditions.get('text')
        
        # Model prediction
        pred = self.wan_t2v.model(
            noisy_frames,
            t,
            context=[text_context] if text_context is not None else None,
            seq_len=self._compute_seq_len(noisy_frames.shape),
            extra_conditions=conditions
        )[0]
        
        # In flow matching, target is the original data
        loss = torch.mean((pred - frames) ** 2 * (1 - t.view(-1, 1, 1, 1, 1)))
        logger.debug(f"Flow matching loss: {loss.item()}")
        
        return loss
